test/e2e/src/pom/autoplayOnScrollPage.ts

```diff
@@ -0,0 +1,24 @@
+import { Page } from '@playwright/test';
+import { VideoComponent } from '../../components/videoComponent';
+import { BasePage } from './BasePage';
+const AUTOPLAY_ON_SCROLL_PAGE_VIDEO_SELECTOR = '//*[@id="player_html5_api"]';
+
+/**
+ * Video player examples autoplay on scroll page object
+ */
+export class AutoplayOnScrollPage extends BasePage {
+  public autoplayOnScrollVideoComponent: VideoComponent;
+
+  constructor(page: Page) {
+    super(page);
+    this.autoplayOnScrollVideoComponent = new VideoComponent(page, AUTOPLAY_ON_SCROLL_PAGE_VIDEO_SELECTOR);
+  }
+
+  /**
+   * Scrolls the page until the video element is visible using scrollIntoViewIfNeeded method.
+   * This action ensures the autoplay behavior is triggered when the video comes into view.
+   */
+  async scrollToVideoElement() {
+      await this.page.locator(AUTOPLAY_ON_SCROLL_PAGE_VIDEO_SELECTOR).scrollIntoViewIfNeeded();
```

Comment:
you don't need it, you can use something like autoplayOnScrollVideoComponent. scrollIntoViewIfNeeded() in the test.

Reply:
Added BaseComponent and modified the videoComponent to extend it. Removed the scrollToVideoElement function and used scrollIntoViewIfNeeded() in the test.